tangelo/toolboxes/molecular_computation/fno.py

```diff
@@ -0,0 +1,415 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module containing datastructures for interfacing with the Frozen Natural
+Orbitals (FNOs), to automatically truncate the virtual orbital space.
+
+Reference: arXiv:2002.07901
+"""
+
+from typing import List, Sequence, Tuple, Union
+import warnings
+
+import numpy as np
+
+from tangelo.algorithms.classical import MP2Solver
+from tangelo.toolboxes.molecular_computation.molecule import SecondQuantizedMolecule
+
+
+class FNO():
+    """Class to interface with the Frozen Natural Orbitals protocol, that aims
+    at reducing the computational cost of a molecular problem by truncating
+    the virtual space. In general, the virtual orbitals are ranked according to
+    their MP2 occupancies, and selected with a given threshold. They are also
+    transformed to the FNO basis, using the eigenvectors of the MP2
+    virtual-virtual density matrix.
+
+    Attributes:
+            sec_mol (SecondQuantizedMolecule): Self-explanatory.
```

Comment:
could we `sq_mol` or `sqmol` in this file to be more consistent with the rest of our codebase please?